btsieve/src/bitcoin/mod.rs

```diff
@@ -6,3 +6,161 @@ pub use self::{
     block_processor::{check_block_queries, check_transaction_queries},
     queries::{BlockQuery, TransactionQuery},
 };
+use crate::{AddBlock, Bitcoin};
+use bitcoin_support::{MinedBlock, Sha256dHash};
+
+impl AddBlock<MinedBlock> for Bitcoin {
+    fn add_block(&mut self, block: MinedBlock) {
+        if self.0.nodes.contains(&block) {
+            return log::warn!("Block already known {:?} ", block);
+        }
+        match self.find_predecessor(&block) {
+            Some(prev) => {
+                self.0.vertices.push((
```

Comment:
You went further than needed, the storage of links between blocks is out of scope :)